### DISCUSSION TITLE
"Dusun DSGW-210-HA” gateway Zigbee and Z-Wave to Ethernet/WiFi appliance hardware for Home Assistant Operating System?

### DISCUSSION THREAD

**Discussion by Hedda** [home-assistant/operating-system#2652] [Ideas]
Linking to discussion thread in Home Assistant community forum about "**Dusun DSGW-210-HA**”, a hardware appliance that ships with Debian for ARM64 and Home Assistant Supervised pre-installed, so question is if it could run Home Assistant Operating System instead?

https://community.home-assistant.io/t/dusun-dsgw-210-ha-gateway-zigbee-and-z-wave-to-ethernet-wifi-appliance-pre-installed-with-home-assistant/593416

If this is good hardware maybe Nabu Casa could rebrand it as “**_Home Assistant Yellow Lite_**” as a low-cost entry model for beginners? 😉 

![image](https://github.com/home-assistant/operating-system/assets/6320001/9d413525-aee4-4d68-a2b5-d4e99a96948d)

![image](https://github.com/home-assistant/operating-system/assets/6320001/0346a796-5289-4ccb-b759-a603c1713fcd)

They an /ODM so also look to make a generic model for OEM rebranding called Dunsun DSGW-230 which features PoE (Power-over-Ethernet):

* https://www.dusuniot.com/product/dsgw-230-gigabit-poe-gateway-ceiling-mountable/

  * https://www.dusuniot.com/product-specification/dsgw-230-multiprotocol-gateway-edge-computing/

All DSGW-210 and DSGW-230 models looks to be based on Rockchip RK3328 SoC so thinking it should perform better than [ASUS Tinker Board S](https://tinker-board.asus.com/product/tinker-board-s.html) as has somewhat similar specifications. (Seeing in some general tests that RK3328 was a little more 30% faster compared to RK3288 when both SoCs were set to run at 1.5GHz).

Product links and documentation:
* https://www.dusuniot.com/product/dsgw-210-home-assistant-zigbee-gateway/
* https://www.dusuniot.com/product-specification/dsgw-210-ha-home-assistant-gateway/
* https://support.dusuniot.com/hc/en-us/articles/13047435476251-DSGW-210-HA
* https://www.dusuniot.com/blog/how-to/how-to-install-application-on-dsgw-210/
* https://www.facebook.com/dusuniot/posts/introducing-our-newest-productdsgw-210-ha-a-linux-debian-gateway-with-home-assis/901899447427791/
*  https://www.linkedin.com/company/dusun-electron-ltd/posts/?feedView=all

FCC report:
* https://fcc.report/FCC-ID/2AUXBDSGW-210/

Videos:
* https://www.youtube.com/watch?v=7hoH7kwfY4A&ab_channel=DusunIoT
* https://www.youtube.com/watch?v=YsFIIBLoDj0&ab_channel=DusunIoT
* https://www.youtube.com/watch?v=56eQZsUJOSw&ab_channel=DusunIoT

Note that the DSGW-210 series from DusunIoT does not seem to be meant to be sold directly to end-users as-is and is instead designed to be sold as a white-label product for rebranding so it is meant to be bought by a company and customized before resold under a new logo.

* https://www.dusuniot.com/product/dsgw-210-rk3328-iot-gateway-hub-controller-bridge-hardware/

They have several different model variants of Dusun DSGW-210 with different radios and other specifications. This specific DSGW-210-HA” model ships with EFR32MG21 radio module for Zigbee/Thread + ZGM130S Z-Wave 700 series and EFR32BG22 for Bluetooth BLE radios.

Apparently, Andrew Howe from [The Smarthome Book](https://www.thesmarthomebook.com/) did test an older pre-production sample of this product for pre-review more than a year ago (29/04/2022), though seeing as the product at that time actually did not look to have been properly launched and advertised until more recently we can perhaps assume that his test was a little outdated, especially how much the installation process of Home Assistant itself has matured a lot since, but he did summarize his review with a comment that "_this product is probably suitable for business to business, and not business to consumer_".

* https://www.thesmarthomebook.com/2022/04/29/dusun-iot-home-assistant-gateway-unbox-set-up-review/

I can currently only find the "Dusun DSGW-210-HA” model for wholesale directly from Dusun/Roombanker on Alibaba (not Aliexpress) which is really meant for distributors for resellers or those wanting to customize logo and packaging for rebranding. See these two links:

* https://www.alibaba.com/product-detail/Dusun-Open-API-Smart-Home-Gateway_1600274606083.html

* https://www.alibaba.com/product-detail/Dusun-Smart-Home-Raspberry-Pi-Grow_1600273967614.html

PS: FYI, looks like the Jeedom SAS team has rebranded a slightly different model of Dusun DSGW-210 or Dusun  DSGW-230 and is reselling it as “**_Jeedom Luna_**”:

* https://www.jeedom.com/fr/box.html
* https://blog.jeedom.com/6842-destination-jeedom-luna/
* https://haade.fr/fr/blog/jeedom-luna-passe-au-crible
* https://shop.domo-supply.com/en/smarthome-hub/2058-jeedom-home-automation-controller-jeedom-luna-z-wave-700-and-zigbee-30.html
* https://www.domadoo.fr/en/smart-home-gateway/6280-jeedom-smart-home-gateway-jeedom-luna-zigbee-30-and-z-wave-700.html

**Comment by Hedda** [comment]
I bet that they are willing to collaborate on manufacturing to make a customized model for Nabu Casa as part of some kind of partnership.

https://partner.home-assistant.io/

https://www.home-assistant.io/blog/2022/07/12/partner-program/

PS: Reading RK3328 SoC (Rockchip RK3300 family) specification it should support 4GB RAM, but they only list 1GB and 2GB models today: 

https://www.dusuniot.com/product/dsgw-210-rk3328-iot-gateway-hub-controller-bridge-hardware/

**Comment by pipiche38** [comment]
Do you have any idea on how we can upgrade the firmware of the Zigbee part. Looks like they are running 6.7.10.0 build 423

**Reply by Hedda** [reply]
Sorry no, but there is a longer discussion thread here with more links -> https://community.home-assistant.io/t/dusun-dsgw-210-ha-zigbee-thread-z-wave-and-ble-gateway-appliance-pre-installed-with-home-assistant-operating-system/593416/

Last I heard anything new from Dusun IoT was in this article on CNX Software which mentioned that Dusun IoT was working on replacing its Google Docs-based documentation with a proper online wiki and GitHub repos. That article might still be a good read:

https://www.cnx-software.com/2023/10/24/dsom-020r-px30-development-board-review-unboxing-benchmarks-and-zigbee-gateway/

They also have some videos on their YouTube channel, see example:

https://www.youtube.com/watch?v=56eQZsUJOSw

But that does not yet to be the case look that they changes their ways as their wiki still link to Google Docs:

- https://wiki.dusuniot.com/iot-gateway-hardware/dsgw-210-rk3328-gateway/overview

and their GitHub account does not look to be active:

- https://github.com/dusuniot

Most of their support still looks to be done via their online community:

https://community.dusuniot.com/

You could also try sening a direct message to thier company representivie that has a registered Home Assistant community account here (but has not been active there in a couple of years) -> https://community.home-assistant.io/u/dusun_iot/summary

Anyway, if it is has the standard Silabs bootloader and running Linux with shell access then I assume you could probably flash it with Nabu Casa's universal-silabs-flasher:

* https://github.com/NabuCasa/universal-silabs-flasher

But be careful since you risk bricking it if the firmware image build so not have the correct pin mappings, see silabs firmware builder:

* https://github.com/darkxst/silabs-firmware-builder

PS: If you have a model that is already running Home Assistant Operating System then you can install the "Advanced SSH & Web Terminal" community add-on (from HACS) which allow you to log in to Home Assistant instance using SSH or by using the Web Terminal.:

- https://github.com/hassio-addons/addon-ssh

**Comment by Hedda** [comment]
FYI, just stumbled on several other slighty newer models of applicances from @dusuniot preloaded with Home Assistant Operating System:

* https://www.dusuniot.com/landing-pages/home-assistant-gateway/

**Dunsun DSGW-120 / DSGW-120V2** (based on Rockchip PX30) looks to be an all-in-one smart home control panel:

* https://www.dusuniot.com/product/dsgw-120-smart-home-control-panel/

  * https://www.youtube.com/watch?v=Xy3cm8QSCME

  Since it uses the Rockchip PX30 it probably uses their DSOM-020R PX30 development board which they announded a few years ago -> https://www.cnx-software.com/2023/10/24/dsom-020r-px30-development-board-review-unboxing-benchmarks-and-zigbee-gateway/

**Dunsun DSGW-290** (based on Rockchip RK3568) looks to be custom appliance with onboard IoT radios and a 4G LTE SIM card slot:

* https://www.dusuniot.com/product/dsgw-290-industrial-edge-computing-gateway-raspberry-pi-4-alternative/

  * https://www.youtube.com/watch?v=hWeC5dke__A

They also updated their HA OS image for the Dunsun DSGW-290 model to include Frigate NVR with Local AI Object Detection on RK3568:

* https://www.dusuniot.com/news/deep-integration-of-home-assistant-and-frigate-nvr/

Not seeing any activity on their @dusuniot GitHub account though so do not know if they are contributing back to upstream in any way?

* https://github.com/dusuniot

Llian Li at Dunsun) also have an inactive company account in the Home Assistant community forum:

* https://community.home-assistant.io/u/dusun_iot/summary

PS: Again, if anyone have any contacts at the Dusun company then maybe suggest to them to at the very least try to follow Sonoff/ITead/Coolkit/eWeLink’s development approach for Sonoff iHost for the better option of porting Home Assistant Operating System in a standardized way to this hardware and publish all of the projects/code/libraries publicly on GitHub to better try to follow the spirit of open-source software and community etos:

* https://github.com/iHost-Open-Source-Project/ha-operating-system

  * https://developers.home-assistant.io/docs/operating-system/debugging/#ssh-access-to-the-host

*  https://community.home-assistant.io/t/ha-over-ihost-open-source-project-is-officially-live-try-it-out-and-join-the-community/878028
